### PR TITLE
[stdlib] Add a negation for 'VectorProtocol' when scalar is signed

### PR DESCRIPTION
**Pull request by rxwei**

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Review by dan-zheng — APPROVED** [review]
-(-1)

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

File: stdlib/public/core/AutoDiff.swift [hunk 1]
```diff
@@ -51,6 +51,12 @@ public extension VectorProtocol {
   }
 }
 
+extension VectorProtocol where VectorSpaceScalar: SignedNumeric {
```
Comment (dan-zheng) [inline stdlib/public/core/AutoDiff.swift]: `public extension`?

```suggestion
public extension VectorProtocol where VectorSpaceScalar: SignedNumeric {
```

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Pull request merged**